core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -602,10 +609,16 @@ private DefaultFunctionResolver utc_timestamp() {
    */
   private DefaultFunctionResolver week(BuiltinFunctionName week) {
     return define(week.getName(),
+        implWithProperties((functionProperties, arg) -> DateTimeFunction.weekOfYearToday(
+            DEFAULT_WEEK_OF_YEAR_MODE,
+            functionProperties.getQueryStartClock()), INTEGER, TIME),
         impl(nullMissingHandling(DateTimeFunction::exprWeekWithoutMode), INTEGER, DATE),
         impl(nullMissingHandling(DateTimeFunction::exprWeekWithoutMode), INTEGER, DATETIME),
         impl(nullMissingHandling(DateTimeFunction::exprWeekWithoutMode), INTEGER, TIMESTAMP),
         impl(nullMissingHandling(DateTimeFunction::exprWeekWithoutMode), INTEGER, STRING),
+        implWithProperties((functionProperties, time, modeArg) -> DateTimeFunction.weekOfYearToday(
+            modeArg,
+            functionProperties.getQueryStartClock()), INTEGER, TIME, INTEGER),
```

Comment:
```suggestion
        implWithProperties(nullMissingHandlingWithProperties((functionProperties, time, modeArg) -> DateTimeFunction.weekOfYearToday(
            modeArg,
            functionProperties.getQueryStartClock())), INTEGER, TIME, INTEGER),
```

Reply:
Fixed in 2517b31f4ed5688dcaa7baa78aa4a025a71a28a4